app/models/cart_validations.rb

```diff
@@ -46,6 +46,30 @@ def check_banned
     errors
   end
 
+  def check_consecutive # rubocop:disable AbcSize
+    errors = []
+    reserver = User.find_by(id: reserver_id)
+    models = get_items.keys
+    models.each do |model|
+      next unless model.max_per_user == 1 && model.max_checkout_length
+      consecutive = Reservation.for_reserver(reserver).for_eq_model(model)
+                    .consecutive_with(start_date, due_date)
+
+      consecutive.each do |c|
+        next unless c.duration + duration > model.max_checkout_length
+        errors << "Reserver has a consecutive reservation (##{c.id}) "\
+          "that passes the duration limit for the model #{model.name}."
+      end
+
+      next unless consecutive.size > 1 &&
+                  duration + consecutive.inject(0) { |a, e| a + e.duration } >
+                  model.max_checkout_length
+      errors << 'Reserver has consecutive reservations that passes the '\
```

Comment:
This could potentially be many reservations, so I'm not sure about including markdown links here